### PR TITLE
completed card cover schema & instance methods

### PR DESCRIPTION
**Pull request by refugedesigns**
### What this PR does (required):
- This is the card cover plugin for adding a cover image to a card

### Screenshots / Videos (front-end only):
- N/A

### Any information needed to test this feature (required):
- create routes and controllers for card cover image to test method instance
- Method instance created: get, create, update, delete

### CODE REVIEW

**Review by ezeilo-su — COMMENTED** [review]
Looks good! Please always link your ticket in any PR you create. (see this [guide])(https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue). Filling the PR template and linking the issue you're addressing is an important requirement here.
Happy coding!

**Comment by refugedesigns** [comment]
> Looks good! Please always link your ticket in any PR you create. See this [guide](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue). Filling the PR template and linking the issue you're addressing is an important requirement here. Happy coding!

PR template filled and linked to create CardCoverPlugin/CardCoverPluginSchema - #52

File: server/models/CardCoverPlugin.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    type: String,
+    required: true
+  }
+}, {timestamps: true}))
+
+CardCoverPluginSchema.methods.get = async function({params}) {
+  if(!params) {
+    throw new Error("Not a valid request!")
+  }
+  return await this.model("CardCoverPlugin").findById(params.coverId)
```
Comment (moffatethan) [inline server/models/CardCoverPlugin.js:20]: So, we're actually going to be enforcing a response structure. 

`{  status code, data/message }`

200 responses should be the following `{ response: 200, data: <Cover data> }`
Anything else -> `{  response: XXX, message: "Any message" }`

Comment (moffatethan) [reply server/models/CardCoverPlugin.js:20]: Please update in all your action handlers.

**Review by moffatethan — CHANGES_REQUESTED** [review]

File: client/src/components/DropZone/DropZone.tsx [hunk 2]
```diff
@@ -13,6 +13,7 @@ interface Props {
 const DropZone = ({ open, onHandleClose, onSetFile }: Props): JSX.Element => {
   const onDrop = useCallback(
     (acceptedFile) => {
+      console.log(acceptedFile);
```
Comment (moffatethan) [inline client/src/components/DropZone/DropZone.tsx:16]: We can remove the console log here

File: server/models/BasePlugin.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      type: String,
+      required: true,
+    },
+    resourceId: {
+      type: Schema.Types.ObjectId,
+      required: true,
+    },
+  },
+  { timestamps: true, collection: "plugins", discriminatorKey: "type" }
+);
```
Comment (moffatethan) [inline server/models/BasePlugin.js:19]: So, Peter has updated his branch so we should update this too

**Review by moffatethan — CHANGES_REQUESTED** [review]

**Review by moffatethan — APPROVED** [review]

**Pull request merged**